python/src/aac/plugins/check/check_aac_impl.py

```diff
@@ -16,242 +16,355 @@
-                location,
-            )
-            if constraint_name not in constraint_results:
-                constraint_results[constraint_name] = []
-            constraint_results[constraint_name].append(result)
-
-    def check_schema_constraint(
-        source_definition: Definition, check_me: Any, check_against
-    ):
-        """Helper method that runs all the constraints for a given schema.
-
-        Args:
-            source_definition (Definition): Source of the check_me field that we are evaluating
-            check_me (Any):                 The field being checked
-            check_against:                  The schema we are comparing the check_me field against
-
-        Returns:
-            (at check level) ExecutionResult: The result of the checks in this helper method
-
-        Raises:
-            LanguageError: If unique schema definition for field type not found for field name
-            LanguageError: If value of field name was something other than a list
-        """
-
-        # make sure we've got a schema
-        context = LanguageContext()
-        if not context.is_aac_instance(check_against, "aac.lang.Schema"):
-            return
-        # collect applicable constraints
-        schema_constraints = []
-        for runner in context.get_plugin_runners():
-            plugin = runner.plugin_definition.instance
-            for constraint in plugin.schema_constraints:
-                if constraint.universal:
-                    schema_constraints.append(
-                        context.create_aac_object(
-                            "SchemaConstraintAssignment",
-                            {"name": constraint.name, "arguments": []},
-                        )
-                    )
-        if check_against.constraints:
-            for constraint_assignment in check_against.constraints:
-                schema_constraints.append(constraint_assignment)
-
-        # Check the check_me against constraints in the defining_schema
-        for constraint_assignment in schema_constraints:
-            constraint_name = constraint_assignment.name
-            constraint_args = constraint_assignment.arguments
-            callback = all_constraints_by_name[constraint_name]
-            result: ExecutionResult = callback(
-                check_me, source_definition, check_against, constraint_args
+    for item in getattr(check_me, field.name):
+        value_to_check = item
+        if value_to_check is not None:
+
+            constraint_results = check_primitive_constraint(
+                field,
+                source_definition,
+                item,
+                field.type[:-2],
+                field_defining_schema[0].instance,
+                all_constraints_by_name,
+                constraint_results
             )
+    return constraint_results
+
+
+def _run_primitive_constraint_not_list(
+    check_me: Any,
+    field, source_definition: Definition,
```

Comment:
```suggestion
    field, 
    source_definition: Definition,
```
And type hint for field